yarpl/include/yarpl/flowable/Flowable.h

```diff
@@ -100,6 +113,16 @@ class Flowable : public virtual Refcounted, public yarpl::enable_get_ref {
 
   Reference<Flowable<T>> observeOn(folly::Executor&);
 
+  template <typename Q>
+  using enableWrapRef =
+      typename std::enable_if<detail::is_t_flowable<Q>::value, Q>::type;
+
+  template <typename Q = T>
+  enableWrapRef<Q> merge() {
+    folly::Function<T(T)> func = [](auto f) { return std::move(f); };
+    return this->flatMap(std::move(func));
```

Comment:
Would it be possible to just copy paste the 'lambda' part into this. compiler might generate an optimized version instead of folly::Function..

this->flatMap([](auto f) { return std::move(f); });

Reply:
A limitation of flatMap right now is that it can't infer return type from the type of the lambda (templates are... hard. I can work on this more in another diff). It's held in flatMap as a folly::Function, so there won't be a perf hit.

Reply:
+1, I think it'd make things a bit cleaner.

Doesn't really matter, but is the `std::move` necessary here?  I can never remember what the rules for RVO are, and which C++ standard made it mandatory.

Reply:
(y) fixed flatMap to deduce types from the lambda its given directly